pkg/ingress/class_loader.go

```diff
@@ -43,9 +46,40 @@ type defaultClassLoader struct {
 	client client.Client
 }
 
+// GetDefaultIngressClass returns the default IngressClass from the list of IngressClasses.
+// If multiple IngressClasses are marked as the default, it returns an error.
+func (l *defaultClassLoader) GetDefaultIngressClass(ctx context.Context) (string, error) {
+	var defaultClass string
+	var defaultClassFound bool
+	ingClassList := &networking.IngressClassList{}
+	if err := l.client.List(ctx, ingClassList); err != nil {
+		return "", fmt.Errorf("failed to fetch ingressClasses, %v", err.Error())
```

Comment:
```suggestion
		return "", fmt.Errorf("fetching ingressClasses: %w", err.Error())
```

Reply:
`err.Error()` is string. I changed the error format based on other errors